game/factions/faction.py

```diff
@@ -0,0 +1,216 @@
+from game.data.doctrine import Doctrine, MODERN_DOCTRINE, COLDWAR_DOCTRINE, WWII_DOCTRINE
+from pydcs_extensions.mod_units import MODDED_VEHICLES, MODDED_AIRPLANES
+
+
+@dataclass
+class Faction:
+
+    # Country used by this faction
+    country: str
+
+    # Nice name of the faction
+    name: str
+
+    # Available aircraft
+    aircrafts: [UnitType]
+
+    # Available awacs aircraft
+    awacs: [UnitType]
+
+    # Available tanker aircraft
+    tankers: [UnitType]
+
+    # Available frontline units
+    frontline_units: [VehicleType]
+
+    # Available artillery units
+    artillery_units: [VehicleType]
+
+    # Infantry units used
+    infantry_units: [VehicleType]
+
+    # Logistics units used
+    logistics_units: [VehicleType]
+
+    # List of units that can be deployed as SHORAD
+    shorads: [str]
+
+    # Possible SAMS site generators for this faction
+    sams: [str]
+
+    # Possible Missile site generators for this faction
+    missiles: [str]
+
+    # Required mods or asset packs
+    requirements: {str: str}
+
+    # possible aircraft carrier units
+    aircraft_carrier: [UnitType]
+
+    # possible helicopter carrier units
+    helicopter_carrier: [UnitType]
+
+    # Possible carrier names
+    carrier_names: [str]
+
+    # Possible helicopter carrier names
+    helicopter_carrier_names: [str]
+
+    # Navy group generators
+    navy_generators: [str]
+
+    # Available destroyers
+    destroyers: [str]
+
+    # Available cruisers
+    cruisers: [str]
+
+    # How many navy group should we try to generate per CP on startup for this faction
+    navy_group_count: int
+
+    # How many missiles group should we try to generate per CP on startup for this faction
+    missiles_group_count: int
+
+    # Whether this faction has JTAC access
+    has_jtac: bool
+
+    # Unit to use as JTAC for this faction
+    jtac_unit: str
+
+    # doctrine
+    doctrine: Doctrine
+
+    # List of available buildings for this faction
+    building_set: [str]
+
+    def __init__(self):
+        self.country = ""
+        self.name = ""
+        self.aircrafts = []
+        self.awacs = []
+        self.tankers = []
+        self.frontline_units = []
+        self.artillery_units = []
+        self.infantry_units = []
+        self.logistics_units = []
+        self.shorads = []
+        self.sams = []
+        self.missiles = []
+        self.requirements = {}
+        self.aircraft_carrier = []
+        self.helicopter_carrier = []
+        self.carrier_names = []
+        self.helicopter_carrier_names = []
+        self.navy_generators = []
+        self.destroyers = []
+        self.cruisers = []
+        self.navy_group_count = 0
+        self.missiles_group_count = 0
+        self.has_jtac = False
+        self.jtac_unit = ""
+        self.doctrine = None
+
+    @classmethod
+    def from_json(cls, json):
+
+        faction = Faction()
+
+        faction.country = json.get("country", "USA")
+        faction.name = json.get("name", "???")
```

Comment:
Any reason to be tolerant of bad data here? I'd just make these two cases errors.